lib/kube/proxy/streamproto/proto.go

```diff
@@ -266,19 +267,10 @@ func (s *SessionStream) Done() <-chan struct{} {
 
 // Close closes the stream.
 func (s *SessionStream) Close() error {
-	if atomic.LoadInt32(&s.closed) == 0 {
-		atomic.StoreInt32(&s.closed, 1)
-
-		err := s.conn.WriteMessage(websocket.CloseMessage, []byte{})
-		if err != nil {
+	if atomic.CompareAndSwapInt32(&s.closed, 0, 1) {
```

Comment:
Spotted this incorrect use of atomics during the migration. A separate load and store is not an atomic operation.

Reply:
This is probably a good change nonetheless but I believe this is fine since `Close()` is never called concurrently. Atomics are only used since other goroutines may read.